src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -607,16 +608,28 @@ static boolean requiresChangeParentUpdateAction(@Nonnull final Category category
         }
     }
 
+    private CompletionStage<Void> fetchAndUpdate(@Nonnull final Category oldCategory,
+                                                 @Nonnull final CategoryDraft newCategory) {
+        final String key = oldCategory.getKey();
+        return categoryService.fetchCategory(key)
+                              .thenCompose(categoryOptional -> {
+                                  if (categoryOptional.isPresent()) {
```

Comment:
`isPresent()` condition is usually used wrong. I'd recommend in this case to use more desired approach:
```java
categoryOptional
    .map(fetchedCategory -> buildUpdateActionsAndUpdate(fetchedCategory, newCategory))
    .orElseGet(() -> {
        handleError(format(UPDATE_FAILED, key, FETCH_ON_RETRY), null);
        return CompletableFuture.completedFuture(null);
    })
```